### ISSUE TITLE
bug/KeyError with PDF partition fast strategy element ID — in old_to_new_mapping[parent_id]

### ISSUE THREAD

**Issue by adieuadieu** [Unstructured-IO/unstructured#2931]
**Describe the bug**
When partitioning [this](https://github.com/Unstructured-IO/unstructured/files/15109052/a1977-backus.pdf) PDF document with the `fast` strategy, the following `KeyError` occurs:

```
{
	"name": "KeyError",
	"message": "'782eec119b3409ea1a0bc8abf8f059ac'",
	"stack": "---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
Cell In[32], line 6
      1 from unstructured.partition.auto import partition
      4 file = \"/Users/marco/Downloads/a1977-backus.pdf\"
----> 6 elements = partition(
      7     filename=file,
      8     strategy=\"fast\"
      9 )

File ~/miniconda3/envs/ai-tasks/lib/python3.11/site-packages/unstructured/partition/auto.py:423, in partition(filename, content_type, file, file_filename, url, include_page_breaks, strategy, encoding, paragraph_grouper, headers, skip_infer_table_types, ssl_verify, ocr_languages, languages, detect_language_per_element, pdf_infer_table_structure, extract_images_in_pdf, extract_image_block_types, extract_image_block_output_dir, extract_image_block_to_payload, xml_keep_tags, data_source_metadata, metadata_filename, request_timeout, hi_res_model_name, model_name, date_from_file_object, starting_page_number, **kwargs)
    421 elif filetype == FileType.PDF:
    422     _partition_pdf = _get_partition_with_extras(\"pdf\")
--> 423     elements = _partition_pdf(
    424         filename=filename,  # type: ignore
    425         file=file,  # type: ignore
    426         url=None,
    427         include_page_breaks=include_page_breaks,
    428         infer_table_structure=infer_table_structure,
    429         strategy=strategy,
    430         languages=languages,
    431         hi_res_model_name=hi_res_model_name or model_name,
    432         extract_images_in_pdf=extract_images_in_pdf,
    433         extract_image_block_types=extract_image_block_types,
    434         extract_image_block_output_dir=extract_image_block_output_dir,
    435         extract_image_block_to_payload=extract_image_block_to_payload,
    436         starting_page_number=starting_page_number,
    437         **kwargs,
    438     )
    439 elif filetype in IMAGE_FILETYPES:
    440     elements = partition_image(
    441         filename=filename,  # type: ignore
    442         file=file,  # type: ignore
   (...)
    453         **kwargs,
    454     )

File ~/miniconda3/envs/ai-tasks/lib/python3.11/site-packages/unstructured/documents/elements.py:575, in process_metadata.<locals>.decorator.<locals>.wrapper(*args, **kwargs)
    573 unique_element_ids: bool = params.get(\"unique_element_ids\", False)
    574 if unique_element_ids is False:
--> 575     elements = assign_and_map_hash_ids(elements)
    577 return elements

File ~/miniconda3/envs/ai-tasks/lib/python3.11/site-packages/unstructured/documents/elements.py:523, in assign_and_map_hash_ids(elements)
    521     if not parent_id:
    522         continue
--> 523     e.metadata.parent_id = old_to_new_mapping[parent_id]
    525 return elements

KeyError: '782eec119b3409ea1a0bc8abf8f059ac'"
}
```

**To Reproduce**
```
from unstructured.partition.auto import partition

file = "/Users/marco/Downloads/a1977-backus.pdf"

elements = partition(filename=file, strategy="fast")
```

**Expected behavior**
Expect that the file is correctly partitioned without throwing a KeyError.

**Environment Info**
`unstructured[pdf,doc,docx,xls,xlsx,ppt,pptx,epub,md,html]==0.13.4`

```
OS version:  macOS-14.4.1-arm64-arm-64bit
Python version:  3.11.9
unstructured version:  0.13.4
unstructured-inference version:  0.7.28
pytesseract version:  0.3.10
Torch version:  2.2.2
Detectron2 is not installed
PaddleOCR is not installed
Libmagic version: file-5.41
magic file from /usr/share/file/magic
LibreOffice version:  ==> libreoffice: 24.2.2
```

**Additional context**
[a1977-backus.pdf](https://github.com/Unstructured-IO/unstructured/files/15109052/a1977-backus.pdf)

**Comment by adieuadieu** [comment]
Update — this error is still reproducible in `unstructured==0.13.4`. I was hoping the changes to deterministic IDs would maybe magically resolve the issue — but no such luck.

**Comment by scanny** [comment]
Fixed by #2940.

**Comment by scanny** [comment]
Fixed in v0.13.5.
